### PR TITLE
VM Tests: Run Image Customizer container.

### PR DESCRIPTION
**Pull request by cwize1**
Add executing the Image Customizer container in Docker to the basic test.

The test suite's Makefile will call the toolkit's Makefile to automatically rebuild the imagecustomizer binary and the container. This avoids the user needing to provide a URL to the container. And it ensures the tests are always running against the code within the checked-out repo.

Note: This change makes extensive use of pytest's fixtures feature. If you are unfamilar with those, it is worth reading the [docs](https://docs.pytest.org/en/stable/how-to/fixtures.html#how-to-fixtures).

---

### **Checklist**
- [x] Tests added/updated
- [x] Documentation updated (if needed)
- [x] Code conforms to style guidelines

### CODE REVIEW

File: test/vmtests/vmtests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+import pytest
+from docker import DockerClient
+
+SCRIPT_PATH = Path(__file__).parent
+TEST_CONFIGS_DIR = SCRIPT_PATH.joinpath("../../../toolkit/tools/pkg/imagecustomizerlib/testdata")
+
+
+def pytest_addoption(parser: pytest.Parser) -> None:
+    parser.addoption("--keep-environment", action="store_true", help="Keep the resources created during the test")
+    parser.addoption("--core-efi-azl2", action="store", help="Path to Azure Linux 2.0 core-efi qcow2 image")
```
Comment (gmileka) [inline test/vmtests/vmtests/conftest.py:22]: why only azl2?

File: test/vmtests/vmtests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        shutil.rmtree(temp_path)
+
+
+@pytest.fixture(scope="function")
+def test_instance_name(request: pytest.FixtureRequest) -> Generator[str, None, None]:
+    instance_suffix = "".join(random.choice(string.ascii_uppercase) for _ in range(5))
+    yield request.node.name + "-" + instance_suffix
+
+
+# pytest has a in-built fixture called tmp_path. But that uses /tmp, which sits in memory.
```
Comment (gmileka) [inline test/vmtests/vmtests/conftest.py:55]: nit: `an`

File: test/vmtests/vmtests/utils/imagecustomizer.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+from pathlib import Path
+
+from docker import DockerClient
+
+from .docker_utils import container_run
+
+
+# Run the containerized version of the imagecustomizer tool.
+def run_image_customizer(
```
Comment (gmileka) [inline test/vmtests/vmtests/utils/imagecustomizer.py:12]: should we expose the partition extraction parameters too?

File: test/vmtests/vmtests/conftest.py (hunk 1 above)
Comment (cwize1) [reply test/vmtests/vmtests/conftest.py:22]: Others will be added in the future. Just keeping it simple for now.

File: test/vmtests/vmtests/utils/imagecustomizer.py (hunk 3 above)
Comment (cwize1) [reply test/vmtests/vmtests/utils/imagecustomizer.py:12]: Probably eventually. But might as well keep it simple for now.

File: test/vmtests/vmtests/conftest.py (hunk 2 above)
Comment (cwize1) [reply test/vmtests/vmtests/conftest.py]: Fixed.

**Review by gmileka — APPROVED** [review]

**Pull request merged**